pyogrio/arrow_bridge.h

```diff
@@ -0,0 +1,109 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// This file is an extract https://github.com/apache/arrow/blob/master/cpp/src/arrow/c/abi.h
+// WARNING: DO NOT MODIFY the content as it would break interoperability !
+
+#pragma once
+
+/*! @cond Doxygen_Suppress */
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define ARROW_FLAG_DICTIONARY_ORDERED 1
```

Comment:
Should probably also add the inclusion guards that are upstream and in the spec:
https://github.com/apache/arrow/blob/74756051c4f6a8b13a40057f586817d56198d4ba/cpp/src/arrow/c/abi.h#L26-L27

Reply:
Ah, good point. I copied it from GDAL instead of Arrow's source, so we should also add those guards in GDAL: https://github.com/OSGeo/gdal/blob/588a2003da0fffa6049ff433dfdb78dd9ff6b533/ogr/ogr_recordbatch.h